frictionless/portals/ckan/adapter.py

```diff
@@ -106,9 +107,33 @@ def write_package(self, package: Package) -> Union[None, str]:
 
                 # upload resources
                 # TODO: See if it's possible to upload only the resources that need to be uploaded
-                for resource in package.resources:
+                for index, resource in enumerate(package.resources):
+                    if resource.path:
+                        resource_filename = resource.path.split("/")[-1]
+                        package.resources[index].path = resource_filename
```

Comment:
I think it might be better to get `descriptor = package.to_descriptor()` first and then do modifications as we don't want to affect initial package in general

Reply:
Also, it's better to use `package.to_descriptor(validate=True)` (it's a new flag) to ensure that the package is valid. For example, we might get into a problem of duplicate resource names:
- table.csv
- data/table.csv
Maybe it will be better to replace names like:
- table.csv -> table.csv
- data/table.csv -> data_table.csv

Reply:
The name collision seems to be handled on CKAN part. It uses `name` of the resource to write the file. So I changed the path rewrite to use `name` .